core/ecschema-editing/src/Merging/ClassMerger.ts

```diff
@@ -124,6 +124,38 @@ export class ClassMerger<TClass extends ECClass> {
     }
   }
 
+  private async mergeCustomAttributes(classKey: SchemaItemKey, changes: Iterable<CustomAttributeContainerChanges>): Promise<SchemaItemEditResults> {
+    for (const customAttributeContainerChange of changes) {
+      for (const change of customAttributeContainerChange.customAttributeChanges) {
+        if (change.changeType === ChangeType.Missing) {
+          const sourceCustomAttribute = change.diagnostic.messageArgs![0] as CustomAttribute;
+          const [schemaName, itemName] = SchemaItem.parseFullName(sourceCustomAttribute.className);
+          const schemaItemKey = new SchemaItemKey(itemName, this.context.sourceSchema.schemaKey.compareByName(schemaName)
+            ? this.context.targetSchema.schemaKey
+            : new SchemaKey(schemaName),
+          );
+          const targetCustomAttribute = await this.context.targetSchema.lookupItem<CustomAttributeClass>(schemaItemKey);
+          if (targetCustomAttribute === undefined) {
+            return { errorMessage: `Unable to locate the custom attribute class ${schemaItemKey.name} in the merged schema.`};
+          }
+
+          const customAttribute = {
+            ...sourceCustomAttribute,
+            className: targetCustomAttribute.fullName,
+          };
+
+          const results = await this.context.editor.entities.addCustomAttribute(classKey, customAttribute);
+          if (results.errorMessage !== undefined) {
+            return {  errorMessage: results.errorMessage };
+          }
+        } else {
+          return { errorMessage: `Changes of Custom Attribute ${customAttributeContainerChange.ecTypeName} on ${classKey.name} merge is not implemented.`};
```

Comment:
This might be a case where we have to be a bit more flexible and not error out when we hit a difference in a custom attribute ... I guess we should see how it goes for now, but keep it in mind

Reply:
I added a backlog item #6311